README.md

```diff
@@ -235,53 +362,113 @@ Classes should start as `final`, and only be changed to allow subclassing if a v
 
-Methods of parameterized types can omit type parameters on the receiving type when they’re identical to the receiver’s. For example:
+**For example:**
 
 ```swift
 struct Composite<T> {
 	…
-	func compose(other: Composite<T>) -> Composite<T> {
-		return Composite<T>(self, other)
+	func compose(other: Composite) -> Composite {
+		return Composite(self, other)
 	}
 }
 ```
 
-could be rendered as:
+**Not**
 
 ```swift
 struct Composite<T> {
 	…
-	func compose(other: Composite) -> Composite {
-		return Composite(self, other)
+	func compose(other: Composite<T>) -> Composite<T> {
+		return Composite<T>(self, other)
 	}
 }
 ```
 
 _Rationale:_ Omitting redundant type parameters clarifies the intent, and makes it obvious by contrast when the returned type takes different type parameters.
 
-#### Use whitespace around operator definitions
+####Operator Definitions
 
-Use whitespace around operators when defining them. Instead of:
+Use whitespace around operators when defining them.
+
+**For example:**
 
 ```swift
-func <|(lhs: Int, rhs: Int) -> Int
-func <|<<A>(lhs: A, rhs: A) -> A
+func <| (lhs: Int, rhs: Int) -> Int
+func <|< <A>(lhs: A, rhs: A) -> A
 ```
 
-write:
+**Not**
 
 ```swift
-func <| (lhs: Int, rhs: Int) -> Int
-func <|< <A>(lhs: A, rhs: A) -> A
+func <|(lhs: Int, rhs: Int) -> Int
+func <|<<A>(lhs: A, rhs: A) -> A
 ```
 
 _Rationale:_ Operators consist of punctuation characters, which can make them difficult to read when immediately followed by the punctuation for a type or value parameter list. Adding whitespace separates the two more clearly.
 
-#### Translations
+####Functions
 
-* [中文版](https://github.com/Artwalk/swift-style-guide/blob/master/README_CN.md)
-* [日本語版](https://github.com/jarinosuke/swift-style-guide/blob/master/README_JP.md)
-* [한국어판](https://github.com/minsOne/swift-style-guide/blob/master/README_KR.md)
\ No newline at end of file
+Specifications around the preferable syntax to use when declaring, and using functions.
+
+#####Declarations
+
+When declaring a function, external parameters are always preferable.  Functions that don't include the first argument in the name should declare a first argument explicitly.
+
+**For example:**
+
+```Swift
+func moveView(view: UIView, toFrame frame: CGRect)
+```
+
+**Not**
+
+```Swift
+func move(view: UIView, _ frame: CGRect)
+```
+
+#####Calling
+
+... some specifications on calling functions
+
+- Avoid declaring large arguments inline
+- For functions with many argument, specify each arg on new line and `)` on final line
+- Use trailing closure syntax for simple functions
+- Avoid trailing closures at the end of functions with many arguments.  (3+)?
+
+
+####Closures
+
+Shorthand argument syntax should only be used in closures that can be understood in a few lines.  In other situations, declaring a variable that helps identify the underlying value is preferred.
+
+**For example:**
+
+```Swift
+doSomethingWithCompletion() { result in
```

Comment:
It works for me, here's a quick snippet:

```Swift
enum Result {
    case One
    case Two
}

func doSomethingWithCompletion(handler: Result -> Void) {
    handler(.One)
}

doSomethingWithCompletion() { result in
    switch result {
    case .One:
        print("one")
    case .Two:
        print("two")
    }
}
```

Could also be:

```Swift
doSomethingWithCompletion { result in
```

(no parentheses)  

Xcode defaults this way, so maybe we should consider this for preferred syntax.